packages/core/src/schema/schema.test.ts

```diff
@@ -1,452 +1,66 @@
-      id: Enum!
-      entities: [Entity!]! @derivedFrom(field: "relatedEntity")
-    }
-  `);
-
-  expect(() => buildSchema(graphqlSchema)).toThrowErrorMatchingInlineSnapshot(
-    `"Related entity id field is not a scalar: RelatedEntity"`
-  );
-});
-
-test("derivedFrom field where id field is valid", () => {
-  const graphqlSchema = buildGraphqlSchema(`
-    type Entity @entity {
-      relatedEntity: RelatedEntity!
-    }
-
-    type RelatedEntity @entity {
-      id: Int!
-      entities: [Entity!]! @derivedFrom(field: "relatedEntity")
-    }
-  `);
-
-  const schema = buildSchema(graphqlSchema);
-  const entity = schema.entities.find((e) => e.name === "RelatedEntity");
-  const derivedFromField = entity?.fields.find(
-    (f): f is DerivedField => f.name === "entities"
-  );
-  expect(derivedFromField?.kind).toBe("DERIVED");
-  expect(derivedFromField?.notNull).toBe(true);
-  expect(derivedFromField?.derivedFromEntityName).toBe("Entity");
-  expect(derivedFromField?.derivedFromFieldName).toBe("relatedEntity");
+import { test } from "vitest";
+
+import { createColumn, createEnum, createSchema } from "./schema";
+
+// This may be in a museum one day
+// We like this one!!
+// const schema = createSchema([
+//   createTable("share")
+//     .addColumn("id", "string")
+//     .addColumn("name", "string")
+//     .addColumn(
+//       "usernames",
+//       "string",
+//       { optional: true, list: true }
+//     ),
+//   createTable("account")
+//     .addColumn("id", "string")
+//     .addColumn("name", "string")
+//     .addColumn("shareId", "string", { references: "share.id" }),
+// ]);
+
+// const schema = createSchema({
+//   enum: createEnum("one", "two", "three"),
+//   share: createColumn("id", "string")
+//     .addColumn("name", "string")
+//     .addColumn("ref", "enum:enum")
+//     .addColumn("usernames", "string", { optional: true, list: true }),
+//   account: createColumn("id", "string")
+//     .addColumn("name", "string")
+//     .addColumn("shareId", "string", { references: "share.id" }),
+// });
+
+// const schema = createSchema({
+//   share: [
+//     createColumn("id", "string"),
+//     createColumn("name", "string"),
+//     createColumn("usernames", "string", { optional: true, list: true }),
+//   ],
+//   account: [
+//     createColumn("id", "string"),
+//     createColumn("name", "string"),
+//     createColumn("shareId", "string", { references: "share.id" }),
+//   ],
+// });
+
+test("create schema", () => {
+  createSchema({
+    name: createColumn("id", "bigint").addColumn("age", "boolean"),
+  });
+});
+
+test("create enum", () => {
+  createSchema({
+    enummm: createEnum("ONE", "TWO", "THREE"),
+    name: createColumn("id", "bigint").addColumn("age", "enum:enummm"),
+  });
+});
+
+test("references", () => {
+  createSchema({
+    Person: createColumn("id", "string").addColumn("age", "int"),
+    Dog: createColumn("id", "string").addColumn("owner", "string", {
+      references: "Person.id",
+    }),
```

Comment:
Ooh, I quite like this style (use object properties as table/enum names). But, it feels weird that `Dog`, which is the name of a table, has `createColumn` next to it. I can see why, but it's still kinda weird.

Just spitballing again, but what do you think about this:

```ts
import { createSchema, createTable, createColumn } from "@ponder/core"

const schema = createSchema({
  Person: createTable({
    id: createColumn("string"),
    age: createColumn("int"),
  }),
  Dog: createTable({
    id: createColumn("string"),
    ownerId: createColumn("string", { references: "Person.id" }),
})
```

Feels maybe a bit verbose. Another spitball:

```ts
import { createSchema, table, enum, column } from "@ponder/core"

const schema = createSchema({
  PetKind: enum(["ONE", "TWO", "THREE"]),
  Person: table({
    id: column("string"),
    age: column("int"),
  }),
  Dog: table({
    id: column("string"),
    kind: column("PetKind"),
    ownerId: column("string", { references: "Person.id" }),
})
```